addon/mixins/osf-model.js

```diff
@@ -7,5 +7,37 @@ import DS from 'ember-data';
 
 export default Ember.Mixin.create({
     links: DS.attr('links'),
-    embeds: DS.attr('embed')
+    embeds: DS.attr('embed'),
+
+    relationshipLinks: DS.attr('links'),
+    dirtyRelationships: Ember.computed('_dirtyRelationships', function() {
+        var dirtyRelationships = this.get('_dirtyRelationships');
+        return Object.keys(dirtyRelationships).map((rel) => {
+            if (rel === 'files' || rel === 'node' || rel === 'user') {
+                return null;
+            }
+            if (dirtyRelationships[rel]) {
+                return rel;
+            }
+            return null;
+        }).filter(Boolean);
+    }),
+    _dirtyRelationships: {},
+    onLoad: Ember.on('ready', function() {
+        this.eachRelationship((rel, meta) => {
+            var _this = this;
```

Comment:
Test suite wasnt happy with ` bind `, and linter complains that ` self ` should be ` _this `